File: lawza-go/mail/mail.go
package mail

import (
    "gopkg.in/gomail.v2"
    "os"
)

func SendOtpEmail(to string, otp string) error {
    m := gomail.NewMessage()
    m.SetHeader("From", os.Getenv("SMTP_USER"))
    m.SetHeader("To", to)
    m.SetHeader("Subject", "Your Lawza OTP")
    m.SetBody("text/plain", "Your OTP code is "+otp)
    d := gomail.NewDialer(os.Getenv("SMTP_HOST"), 587, os.Getenv("SMTP_USER"), os.Getenv("SMTP_PASS"))
    return d.DialAndSend(m)
}

File: lib/mail.ts
import nodemailer from 'nodemailer';

export async function sendOtpEmail(email: string, otp: string) {
  let transporter = nodemailer.createTransport({
    host: process.env.SMTP_HOST!,
    port: Number(process.env.SMTP_PORT!),
    secure: false,
    auth: {
      user: process.env.SMTP_USER!,
      pass: process.env.SMTP_PASS!,
    },
  });
  await transporter.sendMail({
    from: '"Lawza" <no-reply@lawza.pk>',
    to: email,
    subject: "Your Lawza OTP",
    text: `Your OTP is ${otp}`,
    html: `<b>Your OTP is: ${otp}</b>`,
  });
}